src/overlay/Overlay_D3D12.cpp

```diff
@@ -115,23 +214,23 @@ void Overlay::Render(IDXGISwapChain3* pSwapChain)
     DrawImgui(pSwapChain);
 
     const auto bufferIndex = pSwapChain->GetCurrentBackBufferIndex();
-
-    auto& currentFrameContext = m_frameContexts[bufferIndex];
-    currentFrameContext.CommandAllocator->Reset();
+    auto& frameContext = m_frameContexts[bufferIndex];
+    m_pd3dCommandAllocator->Reset();
```

Comment:
I remember using the same allocator for every frame would make the overlay randomly working or not, are you sure about that?

Reply:
Not 100% sure now tbh. I'll double-check just to be sure after lineskips and such.

But tested all the scenarios I could think of (mixing different overlays running at once with this, only some,...)
Seems like there is no issue with how it is currently in my branch.

Reply:
You were correct, there could be an issue if two threads tried to use it at once, which could technically happen.

So fixed it.

Though it also seems like we should probably have separate command lists for each buffer as well...
Looking at how Reshade does it though, seems like we should be good in command lists regard with single list as we have now.